contrib/signet/getcoins.py

```diff
@@ -47,4 +47,14 @@
 except:
     print('Unexpected error when contacting faucet:', sys.exc_info()[0])
     exit()
-print(res.text)
+
+# Display the output as per the returned status code
+if res:
+    # When the return code is in between 200 and 400 i.e. successful
+    print(res.text)
+elif res.status_code == 404:
+    print('The specified faucet URL does not exist. Please check for any server issues/typo.')
+else:
+    print(f'Returned Error Code {res.status_code}\n{res.text}\n')
+    print('Please check the provided arguments for their validity/any possible typo.')
+    print('NOTE: The global faucet is currently rate-limited to 1 request per IP per 24hrs. If that\'s the case, please try again after the time limit!')
```

Comment:
There is no such thing. Error is misleading.

Used the same IP for these 2 transactions:
https://explorer.bc-2.jp/tx/e18da1b1df25376ace8b75fcab9c54c72a601cb52c04568da2b52dfc61ebec18
https://explorer.bc-2.jp/tx/17770112c1939212bcc20903a686335296a7713eb80654ac6ef411a96c747bbd

Just had to open https://signetfaucet.com/ in browser and enter details.

If the error isn't changed, it's a NACK from me.

Reply:
Actually, the faucet only asks for the captcha for the second (or more) time usage from the same IP, not the first time. Hence, the script in its current form cannot be used more than once in a while (as the faucet needs some hours before being used without captcha from the same IP).

That's why it kindof acts like a rate limiter (for the current script). The actual rate limits are 1 btc per hr or 500 requests per week. So now, we can either state explicitly that the script currently supports single time usage only and they can use the browser manually for more or we would have to upgrade the script for captcha (which IMO would not be worth the maintenance). @0xB10C Would love to hear your thoughts

Reply:
> Just had to open https://signetfaucet.com/ in browser and enter details.

This is missing the point. This changes the `getcoins.py` script. You can't assume that because something the browser works is does work the same in the script. The script might even be used somewhere automatically where you can't open a browser.

Furthermore, this is in an else branch and **only printed IF an error occurs**. Additionally, it clearly says **NOTE** informing the user that this might be the reason for failure.

Based on https://github.com/bitcoin/bitcoin/pull/22565#issuecomment-887820382 I think @NikhilBartwal ran into this problem when using the script and I think it's reasonable to mention it if an error occurs.